### PR TITLE
[SYCL] Replace SYCL_DEVICE_FILTER with ONEAPI_DEVICE_SELECTOR

### PR DESCRIPTION
**Pull request by raaiq1**
-Modified tests and config to use ONEAPI_DEVICE_SELECTOR instead of SYCL_DEVICE_FILTER
-Added to test to check for warning message when using setting SYCL_DEVICE_FILTER environment variable

### CODE REVIEW

**Review by kbobrovs — APPROVED** [review]
InvokeSimd part LGTM

**Comment by raaiq1** [comment]
Failure SYCL :: KernelAndProgram/kernel-bundle-merge-options-env.cpp is seen in other PRs

**Comment by raaiq1** [comment]
Ping to @intel/llvm-reviewers-runtime , @intel/dpcpp-l0-pi-reviewers or @intel/dpcpp-tools-reviewers for review

**Review by smaslov-intel — APPROVED** [review]

File: SYCL/DeprecatedFeatures/deprecated_sycl_device_filter.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// TODO: Remove test when SYCL_DEVICE_FILTER is removed
+#include <sycl/sycl.hpp>
+
+int main() {
+  using namespace sycl;
+  platform p{};
+  auto devices = p.get_devices();
+  std::cout << "Passed test\n";
+  return 0;
+}
```
Comment (steffenlarsen) [inline SYCL/DeprecatedFeatures/deprecated_sycl_device_filter.cpp]:
```suggestion
}

```

File: SYCL/DeprecatedFeatures/deprecated_sycl_device_filter.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// RUN: env SYCL_DEVICE_FILTER='*' %t.out &> %t.log
+// RUN: FileCheck %s < %t.log
+//
+// CHECK:      WARNING: The enviroment variable SYCL_DEVICE_FITLER is deprecated.
+// CHECK-SAME: Please use ONEAPI_DEVICE_SELECTOR instead.
+// CHECK-NEXT: For more details, please refer to:
+// CHECK-NEXT: https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md#oneapi_device_selector
+
+//==------------- deprecated_sycl_device_filter.cpp - SYCL 2020 deprecation test
+//--------------==//
```
Comment (steffenlarsen) [inline SYCL/DeprecatedFeatures/deprecated_sycl_device_filter.cpp]:
```suggestion
//==---- deprecated_sycl_device_filter.cpp - SYCL 2020 deprecation test ----==//
```

File: SYCL/FilterSelector/select_device.cpp [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-// RUN: env SYCL_DEVICE_FILTER=cpu %t.out
-// RUN: env SYCL_DEVICE_FILTER=level_zero:gpu %t.out
-// RUN: env SYCL_DEVICE_FILTER=opencl:gpu %t.out
-// RUN: env SYCL_DEVICE_FILTER=cpu,level_zero:gpu %t.out
-// RUN: env SYCL_DEVICE_FILTER=opencl:acc %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="*:*" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="*:cpu" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR=level_zero:gpu %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR=opencl:gpu %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR='*:cpu;level_zero:gpu' %t.out
```
Comment (steffenlarsen) [inline SYCL/FilterSelector/select_device.cpp:6]: Is there some logic behind when `'` is used and when `"` is used?

Comment (raaiq1) [reply SYCL/FilterSelector/select_device.cpp:6]: There's functionally no difference between `'` and `"`. Sorry for the inconsistent usage of between the two

Comment (steffenlarsen) [reply SYCL/FilterSelector/select_device.cpp:6]: Not a big problem so I am fine with it as-is.

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by steffenlarsen** [comment]
@raaiq1 - Regression/device_num.cpp is failing in a number of CI configurations after this change. See for example https://github.com/intel/llvm/actions/runs/3427185876/jobs/5710263986.

**Comment by raaiq1** [comment]
Hi. The regression is caused by ONEAPI_DEVICE_SELECTOR implicitly allowing ESIMD backend even though it shouldn't. There's a patch for it here: https://github.com/intel/llvm/pull/7336

**Pull request merged**